encrypted-media-respec.html

```diff
@@ -605,8 +605,112 @@
+              algorithm in order to avoid repeated requests to the user for similar configurations.
+              </p>
+            </div>
+            <ol>
+              <li>
+                <p>Let <var>requirements</var> be the list [ <code>distinctiveIdentifier</code>, <code>persistentState</code> ].</p>
+              </li>
+              <li>
+                <p>
+                  Let <var>optional requirements</var> be the sublist of <var>requirements</var> where for each element of the sublist,
+                  the value of the corresponding member of <var>candidate configuration</var> is <a def-id="requirement-optional"></a>.
+                </p>
+              </li>
+              <li>
+                <dl class="switch">
+                  <dt>If <var>optional requirements</var> is not empty:</dt>
+                  <dd>
+                    <ol>
+                      <li>
+                        <p>
+                          Let <var>optional requirement combinations</var> be the list of all sublists of <var>optional requirements</var>
+                          ordered by length (longest first) and then by the position in <var>optional requirements</var> of the first element that differs
+                          between two lists (lowest first). (i.e. [ <code>distinctiveIdentifier</code> ] is before [ <code>persistentState</code> ].)
+                        </p>
+                      </li>
+                      <li>
+                        <p>For each <var>optional requirement combination</var> in <var>optional requirement combinations</var>:<p>
+                        <ol>
+                          <li>
+                            <p>
+                            Let <var>modified candidate configuration</var> be a copy of
+                            <var>candidate configuration</var>.
+                            </p>
+                          </li>
+                          <li>
+                            <p>
+                              For each element of <var>optional requirements</var> not in <var>optional requirement combination</var>, set the
+                              corresponding member of <var>modified candidate configuration</var> to <a def-id="requirement-not-allowed"></a>.
+                            </p>
+                          </li>
+                          <li>
+                            <p>
+                            Let <var>supported configuration</var> be the result of executing the
+                            <a def-id="get-supported-configuration-and-consent-algorithm"></a> algorithm
+                            with <var>implementation</var>, <var>modified candidate configuration</var> and <var>origin</var>.
+                            </p>
+                          </li>
+                          <li>
+                            <p>
+                            If <var>supported configuration</var> is not <code>ConsentDenied</code>,
```

Comment:
I _think_ that if a configuration is not supported with some MediaKeyRequirement fields set to "optional" then it is also not supported with those same fields downgraded to "not-allowed", since the check for support considers both with / without the optional item. Only if consent is denied is it worth trying the downgrade, because the original check for support might have returned something that requires the optional item and then consent may have been denied for that.